datalad/core/local/status.py

```diff
@@ -464,7 +464,7 @@ def custom_result_renderer(res, **kwargs):  # pragma: more cover
                 ac.color_word(type_, ac.MAGENTA) if type_ else '')))
 
     @staticmethod
-    def custom_result_summary_renderer(results):  # pragma: more cover
+    def custom_result_summary_renderer(results, **kwargs):  # pragma: more cover
```

Comment:
FWIW, looking at the diff I am already confused on how  it was
```
 def custom_result_summary_renderer(results, action_summary):
```
in `push` and just 
```
def custom_result_summary_renderer(results):
```
here was supposed to work (different signatures), and with addition of `**kwargs` (exemplified by absent docstrings describing expected parameters) it just makes them even more of "dark magic" to figure out what they expect to be passed into them.

If we are to make those signatures "flexible", as one renderer might want `recursive` and another not, may be we should at least aim to list expected kwargs listed explicitly, and have some `**_` or alike to swallow those it doesn't care about but which might be passed, with `**_` vs `**kwargs` to hint that "I do not care about other options".

Reply:
I don't think listing expected arguments in a docstring (of that function) makes a lot of sense. Those functions are called from only one internal place. Furthermore it's not a bout a specific function but all of them. It's an interface - "what do I need to implement". So, may be the renderers should just become classes derived from an abstract class that defines the signatures. That would be a place to document it ... WDYT?

Reply:
recording my thoughts from our discussion during weekly meeting: such interface(s) should probably established/documented within `Interface` class itself.  But my comment (if still to be used) on `**kwargs` vs `**_` stands .  But defining interfaces etc probably shouldn't be a part of this PR.

Also, this `**kwargs` here is really the `**allkwargs` (from https://github.com/datalad/datalad/pull/5738/files#diff-37fba60e98afe87a28f73afa6b5bbc3410bad59901363e26256da5584cec265cR331) which, if I got it right, contains all arguments passed into the call.  
By passing it like it is done in this PR via `**kwargs`, would forbid to have `results` kwarg for an `Interface` `__call__`, and makes it really unclear from just looking at the code -- what are those (hence instigated my comment on docstrings, and discussion on interfaces).

I think a better strategy would be to not `**` them and pass as a dedicated `call_kwargs` argument.  This would then:
- not shadow `results` or any other argument to renderers we ever decide to add
- make it explicit on what is that thing we are passing in